File: segmentController/src/main/java/com/clumob/segment/controller/SegmentController.java
package com.clumob.segment.controller;

import io.reactivex.annotations.NonNull;
import io.reactivex.annotations.Nullable;


/**
 * Created by prashant.rathore on 05/07/18.
 */
public interface SegmentController<VD> {

    public VD getViewData();

    public void onCreate();

    public void restoreState(Storable restorableState);

    public void onStart();

    public void onResume();

    public void onPause();

    public void onStop();

    public void onDestroy();
}

File: segmentController/src/main/java/com/clumob/segment/controller/SegmentController.kt
package com.clumob.segment.controller

/**
 * Created by prashant.rathore on 05/07/18.
 */
interface SegmentController {

    fun onCreate(args: Storable?)
    fun restoreState(restorableState: Storable?)
    fun onStart()
    fun onResume()
    fun onPause()
    fun onStop()
    fun onDestroy()
}